source/features/default-branch-button.tsx

```diff
@@ -39,6 +44,12 @@ async function add(branchSelector: HTMLElement): Promise<void> {
 		</a>
 	);
 
+	if (pageDetect.isRepoFile404() && await is404(url.href)) {
```

Comment:
If it's a 404, then we don't add the link at all. This is what we used to do on the old 404 page too